ibis-server/tests/routers/v2/connector/test_postgres.py

```diff
@@ -710,12 +711,29 @@ async def test_dry_plan(client, manifest_str):
 
 async def test_model_substitute(client, manifest_str, postgres: PostgresContainer):
     connection_info = _to_connection_info(postgres)
+    # Test with catalog and schema in SQL
     response = await client.post(
         url=f"{base_url}/model-substitute",
         json={
             "connectionInfo": connection_info,
             "manifestStr": manifest_str,
-            "sql": 'SELECT * FROM "public"."orders"',
+            "sql": 'SELECT * FROM "test"."public"."orders"',
+        },
+    )
+    assert response.status_code == 200
+    assert (
+        response.text
+        == '"SELECT * FROM \\"my_catalog\\".\\"my_schema\\".\\"Orders\\" AS \\"orders\\""'
+    )
+
+    # Test without catalog and schema in SQL but in headers(x-user-xxx)
```

Comment:
Could you add a test for the case where the header only has `catalog` or only has `schema`?